From: Maximilian Weigand <mweigand@mweigand.net>
Date: Thu, 4 Aug 2022 21:38:02 +0200
Subject: [rockchip_ebc] add a module parameter 'refresh_waveform' that can be
 used to dynamically change the waveform that is used for global refreshes.
 Can also by set via sys-file:

echo 4 > /sys/module/rockchip_ebc/parameters/refresh_waveform
---
 drivers/gpu/drm/rockchip/rockchip_ebc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_ebc.c b/drivers/gpu/drm/rockchip/rockchip_ebc.c
index f78e607..d8e44e6 100644
--- a/drivers/gpu/drm/rockchip/rockchip_ebc.c
+++ b/drivers/gpu/drm/rockchip/rockchip_ebc.c
@@ -196,6 +196,10 @@ static int refresh_threshold = 20;
 module_param(refresh_threshold, int, S_IRUGO|S_IWUSR);
 MODULE_PARM_DESC(refresh_threshold, "refresh threshold in screen area multiples");
 
+static int refresh_waveform = DRM_EPD_WF_GC16;
+module_param(refresh_waveform, int, S_IRUGO|S_IWUSR);
+MODULE_PARM_DESC(refresh_waveform, "refresh waveform to use");
+
 static int split_area_limit = 12;
 module_param(split_area_limit, int, S_IRUGO|S_IWUSR);
 MODULE_PARM_DESC(split_area_limit, "how many areas to split in each scheduling call");
@@ -1164,7 +1168,7 @@ static int rockchip_ebc_refresh_thread(void *data)
 /* 				 * @DRM_EPD_WF_GLD16: Less flashy 16-level grayscale, plus anti-ghosting */
 				// Not sure why only the GC16 is able to clear the ghosts from A2
 				// rockchip_ebc_refresh(ebc, ctx, true, DRM_EPD_WF_GC16);
-				rockchip_ebc_refresh(ebc, ctx, true, default_waveform);
+				rockchip_ebc_refresh(ebc, ctx, true, refresh_waveform);
 			} else {
 				rockchip_ebc_refresh(ebc, ctx, false, default_waveform);
 			}
